controllers/humiocluster_permission_tokens.go

```diff
@@ -0,0 +1,209 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/humio/humio-operator/pkg/helpers"
+
+	"github.com/humio/humio-operator/pkg/kubernetes"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/humio/humio-operator/api/v1alpha1"
+
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	humioapi "github.com/humio/cli/api"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// extractExistingHumioAdminUserID finds the user ID of the Humio user for the admin account, and returns
+// empty string and no error if the user doesn't exist
+func (r *HumioClusterReconciler) extractExistingHumioAdminUserID(config *humioapi.Config, req reconcile.Request, organizationMode string, username string, organization string) (string, error) {
+	if organizationMode == "multi" || organizationMode == "multiv2" {
+		allUserResults, err := r.HumioClient.ListAllHumioUsersMultiOrg(config, req, username, organization)
+		if err != nil {
+			// unable to list all users
+			return "", err
+		}
+		for _, userResult := range allUserResults {
+			if userResult.OrganizationName == "RecoveryRootOrg" {
+				if userResult.SearchMatch == fmt.Sprintf(" | %s () ()", username) {
+					fmt.Printf("Found user ID using multi-organization query.\n")
+					return userResult.EntityId, nil
+				}
+			}
+		}
```

Comment:
Some day we should probably refactor this, so that we fetch `subscription` as part of `userResult` and so we can match on `userResult.Subscription = "ClusterOwner"` instead of whatever the name of the cluster owner organization happens to be. I see this is just moving over whatever "user & token management" from the auth sidecar into the Reconciler, so its fine to stick with this right now, but ideally we'd tidy this up a bit.

Reply:
If single org. mode is active it seems like organization ID is `SINGLE_ORGANIZATION_ID` with `subscription` being `Paying`.